src_c/rect.c

```diff
@@ -695,6 +695,147 @@ pg_rect_collidelistall(pgRectObject *self, PyObject *args)
+    if (obj_with_rect == NULL) {
+        return NULL;
+    }
+
+    SDL_Rect *ret = pgRect_FromObject(obj_with_rect, &temp);
+    if (!ret) {
+        PyErr_SetString(PyExc_TypeError,
+                        "Key function must return rect or rect-like objects");
+    }
+    return ret;
+}
+
+static PyObject *
+pg_rect_collideobjectsall(pgRectObject *self, PyObject *args, PyObject *kwargs)
+{
+    SDL_Rect *argrect;
+    Py_ssize_t size;
+    int loop;
+    PyObject *list, *obj;
+    PyObject *keyfunc = NULL;
+    PyObject *ret = NULL;
+    static char *keywords[] = {"list", "key", NULL};
+
+    if (!PyArg_ParseTupleAndKeywords(args, kwargs, "O|$O:collideobjectsall",
+                                     keywords, &list, &keyfunc)) {
+        return NULL;
+    }
+
+    if (!PySequence_Check(list)) {
+        return RAISE(PyExc_TypeError,
+                     "Argument must be a sequence of objects.");
+    }
+
+    if (keyfunc == Py_None) {
+        keyfunc = NULL;
+    }
+
+    if (keyfunc != NULL && !PyCallable_Check(keyfunc)) {
+        return RAISE(PyExc_TypeError,
+                     "Key function must be callable with one argument.");
+    }
+
+    ret = PyList_New(0);
+    if (!ret) {
+        return NULL;
+    }
+
+    size = PySequence_Length(list);
+    for (loop = 0; loop < size; ++loop) {
+        obj = PySequence_GetItem(list, loop);
+
+        if (!obj) {
+            Py_DECREF(ret);
+            return NULL;
+        }
+
+        if (!(argrect = pgRect_FromObjectAndKeyFunc(obj, keyfunc))) {
+            Py_XDECREF(obj);
+            Py_DECREF(ret);
+            return NULL;
+        }
+
+        if (_pg_do_rects_intersect(&self->r, argrect)) {
+            if (0 != PyList_Append(ret, obj)) {
+                Py_DECREF(ret);
+                Py_DECREF(obj);
+                return NULL; /* Exception already set. */
+            }
+        }
+        Py_DECREF(obj);
+    }
+
+    return ret;
+}
+
+static PyObject *
+pg_rect_collideobjects(pgRectObject *self, PyObject *args, PyObject *kwargs)
+{
+    SDL_Rect *argrect;
+    Py_ssize_t size;
+    int loop;
+    PyObject *list, *obj;
+    PyObject *keyfunc = NULL;
+    static char *keywords[] = {"list", "key", NULL};
+
+    if (!PyArg_ParseTupleAndKeywords(args, kwargs, "O|$O:collideobjects",
+                                     keywords, &list, &keyfunc)) {
+        return NULL;
+    }
+
+    if (!PySequence_Check(list)) {
+        return RAISE(PyExc_TypeError,
+                     "Argument must be a sequence of objects.");
+    }
+
+    if (keyfunc == Py_None) {
+        keyfunc = NULL;
+    }
+
+    if (keyfunc != NULL && !PyCallable_Check(keyfunc)) {
+        return RAISE(PyExc_TypeError,
+                     "Key function must be callable with one argument.");
+    }
+
+    size = PySequence_Length(list);
+    for (loop = 0; loop < size; ++loop) {
+        obj = PySequence_GetItem(list, loop);
+
+        if (!obj) {
+            Py_XDECREF(obj);
```

Comment:
This line is pretty redundant here because `obj` is already `NULL`

Reply:
ok, I see, removed

Reply:
Cool 👍